src/haswell/implementation.cpp

```diff
@@ -572,8 +572,45 @@ simdutf_warn_unused size_t implementation::utf32_length_from_latin1(size_t lengt
   return scalar::latin1::utf32_length_from_latin1(length);
 }
 
-simdutf_warn_unused size_t implementation::utf8_length_from_latin1(const char * input, size_t length) const noexcept {
-  return scalar::latin1::utf8_length_from_latin1(input,length);
+simdutf_warn_unused size_t implementation::utf8_length_from_latin1(const char *input, size_t len) const noexcept {
```

Comment:
It would be better to split up the contributions in distinct PRs. Here you are pushing a change to utf8_length_from_latin1 under a PR that says "count utf8 for AVX512".

Including many changes in one PR makes it hard to keep track. 

I encourage you to use multiple *small* PRs instead.